src/js/core/factories/GridColumn.js

```diff
@@ -806,15 +806,17 @@ angular.module('ui.grid')
         }
       });
     }
+  };
 
-    // Remove this column from the grid sorting, include inside build columns so has
-    // access to self - all seems a bit dodgy but doesn't work otherwise so have left
```

Comment:
Moving this method out of the build columns method seems to work. It's only called in one place and it has the correct context ($scope.col.unsort())